### PR TITLE
ENH (merge?) backend/frontend for numpy

### PR DESCRIPTION
**Pull request by edouardoyallon**

### CODE REVIEW

**Comment by edouardoyallon** [comment]
Do we agree that for this backend, we do use complex numbers as implemented in numpy?

**Comment by lostanlen** [comment]
> Do we agree that for this backend, we do use complex numbers as implemented in numpy?

yes, that would be great

**Comment by eickenberg** [comment]
> Do we agree that for this backend, we do use complex numbers as implemented in numpy?

What property are you talking about specifically? That numpy by default uses double precision? Otherwise it's the same as in pytorch:

try:
`a = (np.arange(10) + 1j * np.arange(10, 20)).astype('complex64')`

Then:
`b = a.view()`
`b.shape = b.shape + (1,)`
`b.dtype='float32'`
`print(b.shape)`

File: kymatio/scattering2d/backend/numpy_backend.py [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+            integer such that x is subsampled by 2**k along the spatial variables.
+        Returns
+        -------
+        res : tensor_like
+            tensor such that its fourier transform is the Fourier
+            transform of a subsampled version of x, i.e. in
+            FFT^{-1}(res)[u1, u2] = FFT^{-1}(x)[u1 * (2**k), u2 * (2**k)]
+    """
+    def __call__(self, input, k):
+        out = input.new(input.size(0), input.size(1), input.size(2) // k, input.size(3) // k, 2)
```
Comment (eickenberg) [inline kymatio/scattering2d/backend/numpy_backend.py]: All this is still torch code, we agree?

**Comment by edouardoyallon** [comment]
I'm trying to find a pure - simple np padding to match torch... and then we're good!

Feel free to comment

**Pull request closed without merging**